pootle/core/search/backends/elastic_search.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+try:
+    from elasticsearch import Elasticsearch
+except:
+    Elasticsearch = None
+
+from ..base import SearchBackend
+
+
+class ElasticSearchBackend(SearchBackend):
+    def __init__(self, config_name):
+        self._setup_settings(config_name)
+        self._es = self._get_es_server()
+        self._create_index_if_missing()
+
+    def _server_setup_and_alive(self):
+        return self._es is not None and self._es.ping()
+
+    def _get_es_server(self):
+        if self._settings is None and Elasticsearch is None:
+            return None
+        return Elasticsearch([{'host': self._settings['HOST'], 'port': self._settings['PORT']}, ])
+
+    def _create_index_if_missing(self):
+        if self._server_setup_and_alive():
+            if not self._es.indices.exists(self._settings['INDEX_NAME']):
+                self._es.indices.create(self._settings['INDEX_NAME'])
+
+    def _is_valuable_hit(self, unit, hit):
+        if str(unit.id) == hit['_id']:
+            return False
+
+        return True
+
+    def search(self, unit):
+        if not self._server_setup_and_alive():
+            return []
+
+        counter = {}
+        res = []
+        language = unit.store.translation_project.language.code
+        es_res = self._es.search(index=self._settings['INDEX_NAME'],
+                                doc_type=language,
+                                body={
+                                    "query": {
+                                        "match": {
+                                            'source': {
```

Comment:
Knowing that ES's query DSL is based in JSON, I'd rather use double quotes for search bodies. Do not mix them, at least.